webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/AccessTokenService.scala

```diff
@@ -50,14 +53,43 @@ object UserAccessRequest {
 }
 
 
-class AccessTokenService @Inject()(webKnossosServer: WebKnossosServer, cache: CacheApi) {
+class AccessTokenService @Inject()(webKnossosServer: WebKnossosServer, cache: SyncCacheApi) {
 
   val AccessExpiration: FiniteDuration = 2.minutes
 
-  def hasUserAccess(token: String, accessRequest: UserAccessRequest): Fox[UserAccessAnswer] = {
+  def validateAccessForSyncBlock[A](accessRequest: UserAccessRequest)(block: => Result)(implicit request: Request[A], ec: ExecutionContext): Fox[Result] =
```

Comment:
Moved here from `TokenSecuredRequest`. Unfortunately, overloading didn’t work (couldn’t disambiguate), so there are separate ones for `block: => Result` and `block: => Future[Result]`